### DISCUSSION TITLE
Using Tekton Tasks in Dagger Pipelines, aka an excuse to hack on Tekton :)

### DISCUSSION THREAD

**Discussion by jpadams** [dagger/dagger#3592] [General]
I was noticing how nicely Tekton Tasks are defined and wondered if Dagger Pipelines could easily use them. 🤔 

Looking at Task YAML in [Tekton Hub](https://hub.tekton.dev/) and then looking at the codebase, it seems we could Unmarshal Tasks and map them to the Dagger Engine API. 😃 

For example, there's a "golang-build" task in the Hub: 🏗️ 
https://hub.tekton.dev/tekton/task/golang-build
https://api.hub.tekton.dev/v1/resource/tekton/task/golang-build/0.3/raw

Seems like we could bring it into golang and then use the Dagger Go SDK to do some magic 🪄 🦄
https://github.com/tektoncd/pipeline/blob/main/pkg/apis/pipeline/v1/task_types.go#L45-L96
https://github.com/tektoncd/pipeline/blob/main/pkg/apis/pipeline/v1/param_types.go#L44-L63

What do people think?

**Comment by marcosnils** [comment]
also related to https://github.com/dagger/dagger/discussions/3358 about bringing GHA actions to Dagger
